airbyte-protocol/protocol-models/src/main/resources/airbyte_protocol/airbyte_protocol.yaml

```diff
@@ -74,49 +74,65 @@ definitions:
     properties:
       state_type:
         "$ref": "#/definitions/AirbyteStateType"
+      stream:
+        "$ref": "#/definitions/AirbyteStreamState"
+      global:
+        "$ref": "#/definitions/AirbyteGlobalState"
       data:
         description: "(Deprecated) the state data"
         type: object
         existingJavaType: com.fasterxml.jackson.databind.JsonNode
-      global:
-        "$ref": "#/definitions/AirbyteStateBlob"
-      streams:
-        type: array
-        items:
-          "$ref": "#/definitions/AirbyteStreamState"
 
   AirbyteStateType:
     type: string
     description: >
       The type of state the other fields represent.
-      If not set, the state data is interpreted as GLOBAL and should be read from the `data` field for backwards compatibility.
-      GLOBAL means that the state should be read from `global` and means that it represents the state for all the streams.
-      PER_STREAM means that the state should be read from `streams`. Each item in the list represents the state for the associated stream.
+      Is set to LEGACY, the state data should be read from the `data` field for backwards compatibility.
+      GLOBAL means that the state should be read from `global` and means that it represents the state for all the streams. It contains one shared
+      state and individual stream states.
+      PER_STREAM means that the state should be read from `stream`. The state present in this field correspond to the isolated state of the
+      associated stream description.
     enum:
       - GLOBAL
-      - PER_STREAM
-
+      - STREAM
+      - LEGACY
   AirbyteStreamState:
     type: object
-    description: "per stream state data"
-    additionalProperties: false
+    additionalProperties: true
+    required:
+      - stream_descriptor
+      - stream_state
```

Comment:
I guess you are speaking of `stream_state`. On the DB, if the state is null we will remove it from the DB, so it won't be present in the message, that's why it is required. Is is some scenario that was not expected.

Reply:
yes. sorry. `stream_state`.

when a connectors wants to reset the state of a stream, I expected it would look like:
```yaml
stream_descriptor:
  name: accounts
stream_state:
  null
```

how were you thinking about it?

Reply:
I re-request a review even if there is no change at the moment.